### PR TITLE
[homework#7]-js-basics by Ilya Ischenko

### PR DESCRIPTION
**Pull request by ilya07092000**

### CODE REVIEW

File: homeworks/ilya-ischenko/07-js-basic/script.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+console.log(gimme([2, 3, 1]));
+
+// task 8 https://www.codewars.com/kata/578553c3a1b8d5c40300037c/train/javascript
+function binaryArrayToNumber(arr) {
+  return parseInt(Number(arr.join('')), 2);
+};
+console.log(binaryArrayToNumber([0, 0, 0, 1]));
+
+// task 9 https://www.codewars.com/kata/585d7d5adb20cf33cb000235/train/javascript
+function findUniq(arr) {
```
Comment (esvyridov) [inline homeworks/ilya-ischenko/07-js-basic/script.js:92]: Please use the same method of finding the unique number that you have used for finding the odd or even number from the previous homework.

Comment (esvyridov) [reply homeworks/ilya-ischenko/07-js-basic/script.js:92]: Current this method has complexity O(2n^2). Let's say the input array has 10 elements. Running through the array takes `n` (`for`). Then we do two `indexOf` and `lastIndexOf`, they takes `n + n`. The result is `n(n + n)`

File: homeworks/ilya-ischenko/07-js-basic/script.js [hunk 2]
```diff
@@ -90,12 +82,22 @@ console.log(binaryArrayToNumber([0, 0, 0, 1]));
 
 // task 9 https://www.codewars.com/kata/585d7d5adb20cf33cb000235/train/javascript
 function findUniq(arr) {
-  for (let i = 0; i < arr.length; i += 1) {
-    if (arr.indexOf(arr[i]) === arr.lastIndexOf(arr[i])) {
-      return arr[i];
+  let firstNums = [arr[0]];
+  let secondNums = [];
+
+  for (let i = 1; i < 3; i += 1) {
```
Comment (esvyridov) [inline homeworks/ilya-ischenko/07-js-basic/script.js:88]: why does it start from 1?

File: homeworks/ilya-ischenko/07-js-basic/script.js [hunk 3]
```diff
@@ -90,12 +82,22 @@ console.log(binaryArrayToNumber([0, 0, 0, 1]));
 
 // task 9 https://www.codewars.com/kata/585d7d5adb20cf33cb000235/train/javascript
 function findUniq(arr) {
-  for (let i = 0; i < arr.length; i += 1) {
-    if (arr.indexOf(arr[i]) === arr.lastIndexOf(arr[i])) {
-      return arr[i];
+  let firstNums = [arr[0]];
```
Comment (esvyridov) [inline homeworks/ilya-ischenko/07-js-basic/script.js:85]: could we please explain this?

Comment (ilya07092000) [reply homeworks/ilya-ischenko/07-js-basic/script.js:85]: I need to divide first 3 numbers in two arrays,  so I push this element to validate next numbers

**Review by esvyridov — APPROVED** [review]

**Pull request merged**